File: Data Structures & Algorithms/eating-bananas/submission-2.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        int l = 1, r = piles.Max();
        int res = r;
        
        while (l <= r) {
            int m = l + ((r - l) / 2);

            int totalTime = 0;
            foreach (int p in piles) {
                totalTime += (int)Math.Ceiling((double)p / m);
            }

            if (totalTime <= h) {
                res = m;
                r = m - 1;
            }  else {
                l = m + 1;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0

        if len(s) == 0:
            return res

        count = defaultdict(int)
        l = 0

        for r in range(len(s)):
            
            count[s[r]] += 1


            while count[s[r]] == 2:
                count[s[l]] -= 1
                l += 1

            res = max(res, r - l + 1)
            
        return res